src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -523,6 +591,34 @@ public void writeHeader( final VariantContextWriter vcfWriter, final SAMSequence
         vcfWriter.writeHeader(makeVCFHeader(sequenceDictionary, defaultToolHeaderLines));
     }
 
+    /**
+     * Selects appropriate active region genotyping engine for given region
+     * @param region Current region of interest
+     * @return Active genotyping engine with correct ploidy setting for given region
+     */
+    private MinimalGenotypingEngine getLocalActiveGenotyper(Locatable region) {
```

Comment:
This seems... slow... this gets called once for effectively every window over the entire genome? and the other one two for all of that active ones? If this list gets long users are in for a world of pain... 

We have a class in HTSJDK called OverlapDetector that is designed for exactly this (detecting overlaps to a whole bunch of intervals efficiently) you could use that in a few ways... choose to do that per-ployidy, or just detect overlap before bothering to check the features which would save you in the typical case but not if somebody enters the ploydiy.

Reply:
In most use cases the list will be very short (e.g. < 5 lines), but I switched to `OverlapDetector` since that seems like a better idea generally